### DISCUSSION TITLE
CurvilinearCoordinateSystem - Projection domain size

### DISCUSSION THREAD

**Discussion by Hkgayau** [CommonRoad/commonroad-drivability-checker#6] [Q&A]
![Screenshot (77)](https://github.com/user-attachments/assets/7459b318-ca3a-4167-8df3-a8cf175d5bc6)

```python
curvilinear_cosy = CurvilinearCoordinateSystem(ref_path, 100000, 0.1, 3)
```
 
As you can see the border of the domain has only extended to one side but the other side stay constantly the same. How can I extend the other side as well? or is it possible to just remove the domain constraints completely?

**Comment by Hkgayau** [comment]
Still looking for a solution to the question, any help would be greatly appreciated!

**Comment by geraldfw** [comment]
Hi @Hkgayau ,

sorry for the very late reply, I somehow missed the posts in the discussion section.

The domain stays constant on the other side, because it is limited due to the high curvature of the reference path. However, we have also repeatedly had problems with the way the projection domain check is done in the current version. Therefore, we are currently finishing a new version of the curvilinear coordinate system code, which will be an own package and not part of the commonroad-drivbaility-checker anymore. In the new version we have made some improvements, and also allow the user to deactivate the projection domain check (currently this is not possible unfortunately without changing the source code). We will probably publish it on Github within the next week, I will notify you again.

Best,
Gerald
